File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    heap = []
    
    for s in scoville:
        heapq.heappush(heap, s)

    while heap[0] < K and len(heap) >= 2:
        a = heapq.heappop(heap)
        b = heapq.heappop(heap)
        if a == 0 and b == 0:
            return -1
        heapq.heappush(heap, a+b*2)
        answer += 1
    
    if heap[0] < K:
        return -1
        
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;
class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for(int i=0;i<scoville.length;i++) {
            pq.offer(scoville[i]);
        }
        while (!pq.isEmpty()) {
            Integer food = pq.poll();
            if (food >= K) {
                break;
            }
            if (pq.size() > 0) {
                Integer food2 = pq.poll();
                pq.offer(food + food2*2);
                answer += 1;
            } else {
                return -1;
            }
        }
        return answer;
    }
}